crates/symbolicator/src/services/symbolication/process_minidump.rs

```diff
@@ -875,15 +874,20 @@ impl SymbolicationActor {
                 break result;
             }
 
-            let missing_modules: Vec<(DebugId, &RawObjectInfo)> =
-                std::mem::take(&mut result.missing_modules)
-                    .into_iter()
-                    .filter_map(|id| modules.get(&id).map(|info| (id, info)))
-                    .collect();
+            let missing_modules: Vec<(&DebugId, &RawObjectInfo)> = result
+                .missing_modules
+                .iter()
+                .filter_map(|id| modules.get(&id).map(|info| (id, info)))
+                .collect();
 
             let loaded_caches = self
                 .load_cfi_caches(scope.clone(), &missing_modules, sources.clone())
                 .await;
+
+            if loaded_caches.is_empty() {
```

Comment:
`load_cfi_caches` internally just `map`s the missing_modules, so this is still not the right check.

Reply:
So what is the right check?  Keep a set of DebugIds for missing modules and if the new set is the same as the old stop iterating?

Reply:
That should already be the case, as it will not re-try missing modules that it already tried loading. Unless that logic somehow broke.
I suggested to pair on this with @loewenheim today, suggesting that we should put in another exit condition *after* trying to load cfi caches, in the case we didn’t load anything.

Right now, I think we do *at least* two passes here. But we could even avoid the second pass if we don’t manage to load any kind of CFI.

Reply:
it's a good point even the 2nd pass could be avoided, that might be a better approach

Reply:
I'm honestly starting to think it may be a better idea to leave this as it is for the time being because we're planning to get rid of this iteration anyway.